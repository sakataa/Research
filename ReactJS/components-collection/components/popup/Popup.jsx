import React from 'react';
import ReactDOM from 'react-dom';
import { DEFAULT_WIDTH, DEFAULT_HEIGHT} from './constants';
import PropTypes from 'prop-types';

export default class Popup extends React.Component {
    constructor(props) {
        super(props);
    };

    static propTypes = {
        width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
        height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
        content: PropTypes.string,
        title: PropTypes.string,
        isShow: PropTypes.bool,
        onClose: PropTypes.func,
        isHtml: PropTypes.bool
    }

    static defaultProps = {
        width: DEFAULT_WIDTH,
        height: DEFAULT_HEIGHT,
        isShow: false,
        isHtml: false
    }

    onCloseHandler = () => {
        this.props.onClose && this.props.onClose();
    }

    render() {
        const { content, title, isShow, width, height, isHtml } = this.props;

        return (<div className={`popup-container ${isShow ? 'show' : 'hide'}`}>
            <div className='popup-block-body'></div>
            <div className='flat-popup' style={{ height: height, width: width }}>
                <div className='popup-title'>
                    <div className="txt-title">{title}</div>
                    <a className="popup-close" href="javascript:;" onClick={this.onCloseHandler}><i className="icon-close"></i></a>
                </div>
                { isHtml
                    ? (<div className='popup-content' dangerouslySetInnerHTML={{ __html: content }} />)
                    : (<div className='popup-content'>{content}</div>)
                }
            </div>
        </div>);
    }
};