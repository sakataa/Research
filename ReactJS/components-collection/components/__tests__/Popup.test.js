import React from 'react'
import renderer from 'react-test-renderer'

import Popup from '../popup/Popup'

describe('Popup ', () => {
    describe('render', () => {
        it('isShow is false - popup component is hidden', () => {
            const props = {
                content: '<div></div>'
            };

            const component = renderer.create(<Popup {...props} />);

            const result = component.toJSON();
        
            expect(result).toMatchSnapshot();
            expect(result.props.className).toContain('hide');
        });

        it('isShow is true - popup component is shown', () => {
            const props = {
                content: '<div></div>',
                isShow: true
            };

            const component = renderer.create(<Popup {...props} />);

            const result = component.toJSON();
        
            expect(result).toMatchSnapshot();
            expect(result.props.className).toContain('show');
        });

        it('isHtml is false - popup component shows content as text', () => {
            const props = {
                content: '<div></div>',
                isShow: true
            };

            const component = renderer.create(<Popup {...props} />);

            const result = component.toJSON();
        
            expect(result).toMatchSnapshot();
        });

        it('isHtml is true - popup component shows content as html', () => {
            const props = {
                content: '<div></div>',
                isShow: true,
                isHtml: true
            };

            const component = renderer.create(<Popup {...props} />);

            const result = component.toJSON();
        
            expect(result).toMatchSnapshot();
        });
    });

    describe('onCloseHandler', () => {
        it('onClick prop is called', () => {
            const props = {
                onClose: jest.fn()
            };
            const component = new Popup(props);

            component.onCloseHandler();

            expect(props.onClose).toHaveBeenCalled();
        });
    });
});