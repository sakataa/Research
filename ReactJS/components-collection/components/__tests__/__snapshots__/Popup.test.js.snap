// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup  render isHtml is false - popup component shows content as text 1`] = `
<div
  className="popup-container show"
>
  <div
    className="popup-block-body"
  />
  <div
    className="flat-popup"
    style={
      Object {
        "height": "55%",
        "width": "55%",
      }
    }
  >
    <div
      className="popup-title"
    >
      <div
        className="txt-title"
      />
      <a
        className="popup-close"
        href="javascript:;"
        onClick={[Function]}
      >
        <i
          className="icon-close"
        />
      </a>
    </div>
    <div
      className="popup-content"
    >
      &lt;div&gt;&lt;/div&gt;
    </div>
  </div>
</div>
`;

exports[`Popup  render isHtml is true - popup component shows content as html 1`] = `
<div
  className="popup-container show"
>
  <div
    className="popup-block-body"
  />
  <div
    className="flat-popup"
    style={
      Object {
        "height": "55%",
        "width": "55%",
      }
    }
  >
    <div
      className="popup-title"
    >
      <div
        className="txt-title"
      />
      <a
        className="popup-close"
        href="javascript:;"
        onClick={[Function]}
      >
        <i
          className="icon-close"
        />
      </a>
    </div>
    <div
      className="popup-content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<div></div>",
        }
      }
    />
  </div>
</div>
`;

exports[`Popup  render isShow is false - popup component is hidden 1`] = `
<div
  className="popup-container hide"
>
  <div
    className="popup-block-body"
  />
  <div
    className="flat-popup"
    style={
      Object {
        "height": "55%",
        "width": "55%",
      }
    }
  >
    <div
      className="popup-title"
    >
      <div
        className="txt-title"
      />
      <a
        className="popup-close"
        href="javascript:;"
        onClick={[Function]}
      >
        <i
          className="icon-close"
        />
      </a>
    </div>
    <div
      className="popup-content"
    >
      &lt;div&gt;&lt;/div&gt;
    </div>
  </div>
</div>
`;

exports[`Popup  render isShow is true - popup component is shown 1`] = `
<div
  className="popup-container show"
>
  <div
    className="popup-block-body"
  />
  <div
    className="flat-popup"
    style={
      Object {
        "height": "55%",
        "width": "55%",
      }
    }
  >
    <div
      className="popup-title"
    >
      <div
        className="txt-title"
      />
      <a
        className="popup-close"
        href="javascript:;"
        onClick={[Function]}
      >
        <i
          className="icon-close"
        />
      </a>
    </div>
    <div
      className="popup-content"
    >
      &lt;div&gt;&lt;/div&gt;
    </div>
  </div>
</div>
`;
